src/osha/oira/client/browser/conditions.py

```diff
@@ -10,17 +11,16 @@ def terms_changed(self):
         return getattr(self.account, "tc_approved", None) is not None
 
     def __call__(self):
-        self.came_from = self.request.form.get("came_from")
-        if isinstance(self.came_from, list):
-            # If came_from is both in the querystring and the form data
-            self.came_from = self.came_from[0]
+        webhelpers = api.content.get_view(
+            name="webhelpers", context=self.context, request=self.request
+        )
+        self.came_from = webhelpers.get_came_from(
+            default=aq_inner(self.context).absolute_url()
+        )
 
         self.account = get_current_account()
         if self.request.environ["REQUEST_METHOD"] == "POST":
             self.account.tc_approved = CONDITIONS_VERSION
 
-            if self.came_from:
-                self.request.response.redirect(self.came_from)
-            else:
-                self.request.response.redirect(aq_inner(self.context).absolute_url())
+            self.request.response.redirect(self.came_from)
```

Comment:
You should had a return here.
There is no point in calling self.index().
Anyway this can be merged. This was already overlooked in the previous code and it is not a regression introduced by this PR.

Reply:
May be worth changing. I've pushed a new version.